rdmd.d

```diff
@@ -916,3 +900,41 @@ void yap(size_t line = __LINE__, T...)(auto ref T stuff)
     debug stderr.writeln(line, ": ", stuff);
     else stderr.writeln(stuff);
 }
+
+/**
+Used to wrap filesystem operations that should also be logged with the
+yap function. These functions allow the the filenane to be given once
+so the log statement always matches the operation. Using it also guarantees
+you won't forget to include a `yap` alongside any file operation you want
+to be logged.
+*/
+static struct Yap
+{
+    static auto opDispatch(string op, T...)(T args)
```

Comment:
`op` as a parameter name is better suited for operator overloading (`opUnary` / `opBinary`); consider `fn` / `fun` OSLT.

Reply:
ok, switched it